packages/lib/src/interactions/hooks.ts

```diff
@@ -189,22 +185,23 @@ export function useInteraction(id: string, value: Interaction) {
 }
 
 export function useModifierKeyPressed(
-  modifierKey: ModifierKey | undefined
+  modifierKeys: ModifierKey[] | undefined = []
 ): boolean {
-  const [isPressed, togglePressed] = useToggle(true);
-
-  useKeyboardEvent(modifierKey, () => togglePressed(), [], { event: 'keyup' });
-  useKeyboardEvent(
-    modifierKey,
-    () => {
-      // `keydown` is fired repeatedly, so make sure we toggle only once
-      if (!isPressed) {
-        togglePressed();
-      }
-    },
-    [],
-    { event: 'keydown' }
-  );
-
-  return isPressed;
+  const map = useMap<ModifierKey, boolean>(modifierKeys.map((k) => [k, true]));
+
+  useEventListener(window, 'keyup', (event: KeyboardEvent) => {
+    const { key: pressedKey } = event;
+    if (isModifierKey(pressedKey) && map.has(pressedKey)) {
+      map.set(pressedKey, false);
+    }
+  });
```

Comment:
My first thought was to use a loop to call `useKeyboardEvent` for each `modifierKey` but this breaks the rule of hooks.

I therefore had to reimplement this with a `useEventListener`. Perhaps I should attach it to the `Canvas` rather than the `window` ?